app/src/main/java/edu/nextstep/camp/calculator/MainActivity.kt

```diff
@@ -13,17 +14,71 @@ class MainActivity : AppCompatActivity() {
-        binding.button3.setOnClickListener { binding.textView.text = "3" }
-        binding.button4.setOnClickListener { binding.textView.text = "4" }
-        binding.button5.setOnClickListener { binding.textView.text = "5" }
-        binding.button6.setOnClickListener { binding.textView.text = "6" }
-        binding.button7.setOnClickListener { binding.textView.text = "7" }
-        binding.button8.setOnClickListener { binding.textView.text = "8" }
-        binding.button9.setOnClickListener { binding.textView.text = "9" }
+        setupNumberButtonListener()
+        setupArithmeticButtonListener()
+        setupFunctionButtonListener()
+    }
+
+    private fun setupNumberButtonListener() = with(binding) {
+        button0.setOnClickListener { textView.text = "${textView.text}0" }
+        button1.setOnClickListener { textView.text = "${textView.text}1" }
+        button2.setOnClickListener { textView.text = "${textView.text}2" }
+        button3.setOnClickListener { textView.text = "${textView.text}3" }
+        button4.setOnClickListener { textView.text = "${textView.text}4" }
+        button5.setOnClickListener { textView.text = "${textView.text}5" }
+        button6.setOnClickListener { textView.text = "${textView.text}6" }
+        button7.setOnClickListener { textView.text = "${textView.text}7" }
+        button8.setOnClickListener { textView.text = "${textView.text}8" }
+        button9.setOnClickListener { textView.text = "${textView.text}9" }
+    }
+
+    private fun setupArithmeticButtonListener() = with(binding) {
+        buttonPlus.setOnClickListener {
+            if (hasOperand()) {
+                textView.text = "${textView.text} + "
+            }
+
+        }
+        buttonMinus.setOnClickListener {
+            if (hasOperand()) {
+                textView.text = "${textView.text} - "
+            }
+
+        }
+        buttonMultiply.setOnClickListener {
+            if (hasOperand()) {
+                textView.text = "${textView.text} * "
+            }
+        }
+        buttonDivide.setOnClickListener {
+            if (hasOperand()) {
+                textView.text = "${textView.text} / "
+            }
+        }
+    }
+
+    private fun hasOperand(): Boolean = with(binding) {
+        val input = textView.text
+        return input.isNotBlank() && input.last() != ' '
+    }
+
+    private fun setupFunctionButtonListener() = with(binding) {
+        buttonDelete.setOnClickListener {
+            val text = textView.text
+            if (text.isBlank()) return@setOnClickListener
+            textView.text = text.dropLast(
+                text.last().toString()
+                    .toIntOrNull()?.let { 1 } ?: 3
+            )
+        }
+        buttonEquals.setOnClickListener {
+            if (textView.text.isBlank()) return@setOnClickListener
+            runCatching {
+                textView.text = calculate(textView.text.toString()).toString()
```

Comment:
```suggestion
                textView.text = Calculator.calculate(textView.text.toString()).toString()
```
지금처럼 import를 이용하면 함수만 호출할 수 있으나, 이 함수를 직접 타고가지 않는 이상 MainActivity 내부에 구현된 함수로 오해하기가 쉽습니다.
명시적으로 Calculator를 통해서 함수를 호출하도록 변경해주시는 것이 좋습니다.

Reply:
8992c8b 커밋에서 반영하였습니다.
무조건 static import 를 한다고 깔끔한게 아니군요.
가독성에 관한 관점을 제시해주셔서 감사합니다.